src/gpg_ux_nbgl.c

```diff
@@ -0,0 +1,1116 @@
+             sizeof(G_gpg_vstate.menu),
+             "%s\nID: %x / %d",
+             name,
+             serial,
+             G_gpg_vstate.slot + 1);
+
+    nbgl_useCaseHomeExt(APPNAME,
+                        &C_gpg_64px,
+                        G_gpg_vstate.menu,
+                        true,
+                        "Select Slot",
+                        ui_menu_slot_action,
+                        ui_menu_settings,
+                        app_quit);
+}
+
+//  -----------------------------------------------------------
+//  ------------------------ SLOT UX --------------------------
+//  -----------------------------------------------------------
+
+enum {
+    TOKEN_SLOT_SELECT = FIRST_USER_TOKEN,
+    TOKEN_SLOT_DEF,
+};
+
+enum {
+    SLOT_PAGE_SELECT,
+    SLOT_PAGE_DEFAULT,
+    SLOT_PAGE_NB,
+};
+
+static bool slot_nav_cb(uint8_t page, nbgl_pageContent_t* content) {
+    static char* names[GPG_KEYS_SLOTS] = {0};
+    static char text[GPG_KEYS_SLOTS][32];
+    bool ret = false;
+    uint32_t slot;
+
+    memset(content, 0, sizeof(nbgl_pageContent_t));
+    switch (page) {
+        case SLOT_PAGE_SELECT:
+            for (slot = 0; slot < GPG_KEYS_SLOTS; slot++) {
+                snprintf(text[slot],
+                         sizeof(text[slot]),
+                         "Slot %d %s",
+                         (slot + 1),
+                         (N_gpg_pstate->config_slot[1] == slot) ? "[default]" : "");
+                names[slot] = text[slot];
+            }
+            content->type = CHOICES_LIST;
+            content->choicesList.names = (const char* const*) names;
+            content->choicesList.localized = false;
+            content->choicesList.nbChoices = GPG_KEYS_SLOTS;
+            content->choicesList.initChoice = G_gpg_vstate.slot;
+            content->choicesList.token = TOKEN_SLOT_SELECT;
+            ret = true;
+            break;
+        case SLOT_PAGE_DEFAULT:
+            content->type = INFO_BUTTON;
+            content->infoButton.text = "Set default slot";
+            snprintf(G_gpg_vstate.menu,
+                     sizeof(G_gpg_vstate.menu),
+                     "Slot %d",
+                     (G_gpg_vstate.slot + 1));
+            content->infoButton.buttonText = G_gpg_vstate.menu;
+            content->infoButton.buttonToken = TOKEN_SLOT_DEF;
+            ret = true;
+            break;
+    }
+    return ret;
+}
+
+static void slot_ctrl_cb(int token, uint8_t index) {
+    switch (token) {
+        case TOKEN_SLOT_SELECT:
+            if (index != G_gpg_vstate.slot) {
+                G_gpg_vstate.slot = index;
+                G_gpg_vstate.kslot = (gpg_key_slot_t*) &N_gpg_pstate->keys[G_gpg_vstate.slot];
+                gpg_mse_reset();
+                ui_CCID_reset();
+            }
+            break;
+        case TOKEN_SLOT_DEF:
+            nvm_write((void*) (&N_gpg_pstate->config_slot[1]), &G_gpg_vstate.slot, 1);
+            ui_menu_slot_action();
+            break;
```

Comment:
I see that's work the same on nanox, but I don't get why when selecting the default you don't have similar actionas when selecting something else...

Reply:
This is not the same control. With the App, you have a default slot, which will be selected at startup of the App. In parallel, you have the current selected slot (volatile configuration). You can then change the current slot, just for this session, but prefer another one ass the default slot.

First, you select the current slot (and then, in the actions, the new key slot is fully configured).
In requested, you can go on the 2nd page, and use  "Set default" button, and in that case, we just write the current config (selected on the 1st page) in the nvram.